### ISSUE TITLE
footways and service highways terminating near buildings being marked as errors 

### ISSUE THREAD

**Issue by skorasaurus** [mvexel/remapatron#2]
For the current Maproulette challenge of unconnected highways in the US, I've found a handful of them that are highway=service or highway=footway and the highway in question is close (with 15 meters usually), but not connected to a building (building=*). 

I am not sure and after consulting users in IRC once, I'm don't know if these situations are actual connectivity errors that should be fixed (so that the highway connects to the building - I don't know if there's a more common practice whether to connect them or not) or these are false positives that could be removed from maproulette by editing the sql in maproulette. 

One example would be: http://www.openstreetmap.org/?lat=42.831547&lon=-89.509279&zoom=18

**Comment by mvexel** [comment]
Aware of this problem, we can pretty easily filter them out.  This is still relevant for the running challenge, what do you say we filter footways / paths / track at least? @oldtopos thoughts?

**Comment by oldtopos** [comment]
My concern WRT simply dropping these way classes is how they get QAd and rectified.  The attached appeared in the last two days (and I've already cleared some of them, see attached image) and will likely degrade further if not attended.  (Disclaimer:  I do not suggest the editor(s) is/are of bad intent)

![Screen Shot 2013-01-27 at 8 25 20 PM](https://f.cloud.github.com/assets/2193065/101965/f82c140e-6902-11e2-9190-d0c76af7bed7.png)

If these ways belong in the OSM Planet, should they not be properly connected with the motor vehicle/cycling navigable ways (and which eath other) to improve pedestrian (or multi-modal) routing ?  If the data is important enough to keep, does it not deserve community attention to improve/maintain its quality ?

The error query is considering only ways with a highway tag, so the above report is flagging the very short way segment, where the dangling way end is within 5 meters of a crossing/intersection with another way having a highway tag (such as a service way/driveway crossing a footway); or, in some cases, dangling service way ends that are close to one another, as in two parallel driveways between two buildings. 

What I have been doing with these situations is tagging the dangling way end nodes with noexit=yes, which the error check interprets as a signal that the database geometry properly represents ground truth.  I have also connected the way to the building if it appears an entrance is likley, which also clears the error.

There are other situations appropriate for this technique, such as mass transit platforms, stairs and ways that simple end as they descend underground, and so on.

**Comment by mvexel** [comment]
Agreed that we're going to filter out a few real issues when we choose to exclude certain highway types, but on the other hand, the number of false positives may put people off. 
The image just shows me some red dots, could you elaborate what these represent in this context? Why are these valid issues? 
If they are, could we capture these in a separate challenge?

**Comment by oldtopos** [comment]
The dots are highway=path class ways that are not connected; ie way ends are near/over other ways and are not sharing nodes.

No one shared feedback at the Sunnyvale Edit-a-thon indicating the connectivity challenge contained significant false positives (loosely defined at present, IMHO), but that lack is not conclusive.

What other data is available to develop a more empirical metric ?
